py/desispec/scripts/reformat_proctables.py

```diff
@@ -0,0 +1,177 @@
+"""
+desispec.scripts.updateexptables
+================================
+
+"""
+import os
+import glob
+import sys
+import numpy as np
+import re
+import time
+from astropy.table import Table
+
+from desispec.io.meta import findfile
+from desispec.workflow.proctable import get_processing_table_column_defs
+from desispec.workflow.utils import define_variable_from_environment, listpath, \
+                                    pathjoin
+from desispec.workflow.tableio import write_table, load_table
+from desispec.scripts.exposuretable import create_exposure_tables
+
+
+
+def update_processing_tables(nights=None, night_range=None, orig_filetype='csv',
+                           out_filetype='csv', dry_run=False):
+    """
+    Generates updated processing tables for the nights requested. Requires
+    a current processing table to exist on disk.
+
+    Args:
+        nights: str, int, or comma separated list. The night(s) to generate
+                                                   procesing tables for.
+        night_range: str. comma separated pair of nights in form
+                          YYYYMMDD,YYYYMMDD for first_night,last_night
+                          specifying the beginning and end of a range of
+                          nights to be generated. first_night and last_night are
+                          inclusive.
+        orig_filetype: str. The file extension (without the '.') of the exposure
+                            tables.
+        out_filetype: str. The file extension for the outputted exposure tables
+                           (without the '.').
```

Comment:
Given that you've already implemented it, this may not matter, but what's the motivation for orig_filetype and out_filetype?  Do we have any processing tables that aren't *.csv?

Reply:
I adapted the parser from `desi_reformat_exptables` which is much older. In the past exposure_tables and processing_tables were less mature and the file type was specified (with a default of csv). At this point, they are fundamental enough that changing the filetype is unlikely, but I kept it since I didn't see harm in allowing extra functionality.